streams/shared/src/main/scala/zio/stream/ZStreamChunk.scala

```diff
@@ -141,36 +141,49 @@ class ZStreamChunk[-R, +E, +A](val chunks: ZStream[R, E, Chunk[A]]) { self =>
   final def flattenChunks: ZStream[R, E, A] = chunks.flatMap(ZStream.fromChunk)
 
   /**
-   * Executes an effectful fold over the stream of values.
+   * Executes a pure fold over the stream of values - reduces all elements in the stream to a value of type `S`.
```

Comment:
I would like to suggest that we drop the fold variants that operate on chunks here entirely, because they're accessible by just doing something like `stream.chunks.foldM`. What do you think @kamilkloch?

Instead of those, let's expand the methods on `ZStreamChunk` to include parallels to all the fold variants on `ZStream`. These methods should ideally delegate to the folds on `Chunk` where possible.

Reply:
@iravid Please take a look. I had to force-push the changes due to conficlts.